predicators/approaches/llm_open_loop_approach.py

```diff
@@ -135,63 +135,11 @@ def _llm_prediction_to_option_plan(
-        # We assume the LLM's output is such that each line contains a
-        # option_name(obj0:type0, obj1:type1,...).
-        options_str_list = llm_prediction.split('\n')
-        for option_str in options_str_list:
-            option_str_stripped = option_str.strip()
-            option_name = option_str_stripped.split('(')[0]
-            # Skip empty option strs.
-            if not option_str:
-                continue
-            if option_name not in option_name_to_option.keys() or \
-                "(" not in option_str:
-                logging.info(
-                    f"Line {option_str} output by LLM doesn't "
-                    "contain a valid option name. Terminating option plan "
-                    "parsing.")
-                break
-            option = option_name_to_option[option_name]
-            # Now that we have the option, we need to parse out the objects
-            # along with specified types.
-            typed_objects_str_list = option_str_stripped.split('(')[1].strip(
-                ')').split(',')
-            objs_list = []
-            malformed = False
-            for i, type_object_string in enumerate(typed_objects_str_list):
-                object_type_str_list = type_object_string.strip().split(':')
-                # We expect this list to be [object_name, type_name].
-                if len(object_type_str_list) != 2:
-                    logging.info(f"Line {option_str} output by LLM has a "
-                                 "malformed object-type list.")
-                    malformed = True
-                    break
-                object_name = object_type_str_list[0]
-                type_name = object_type_str_list[1]
-                if object_name not in obj_name_to_obj.keys():
-                    logging.info(f"Line {option_str} output by LLM has an "
-                                 "invalid object name.")
-                    malformed = True
-                    break
-                obj = obj_name_to_obj[object_name]
-                # Check that the type of this object agrees
-                # with what's expected given the ParameterizedOption.
-                if type_name != option.types[i].name:
-                    logging.info(f"Line {option_str} output by LLM has an "
-                                 "invalid type name.")
-                    malformed = True
-                    break
-                objs_list.append(obj)
-            # The types of the objects match, but we haven't yet checked if
-            # all arguments of the option have an associated object.
-            if len(objs_list) != len(option.types):
-                malformed = True
-            if not malformed:
-                option_plan.append((option, objs_list))
+        option_plan_with_cont_params = utils.\
+            parse_model_output_into_option_plan(
+            llm_prediction, objects, self._types, self._initial_options, False)
+        option_plan = [(elem[0], elem[1])
```

Comment:
`[(option, objs) for option, objs, cont_params in option_plan_with_cont_params`]